cli/types/index.d.ts

```diff
@@ -50,6 +50,7 @@ declare namespace Cypress {
   type RequestBody = string | object
   type ViewportOrientation = "portrait" | "landscape"
   type PrevSubject = "optional" | "element" | "document" | "window"
+  type PluginConfig = (on: PluginEvents, config: ConfigOptions) => void

```

Comment:
the return type for plugin config function can return some props from config object or Promise to it or nothing I think